src/TestSuite/Constraint/Email/MailSubjectContains.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Constraint\Email;
+
+use InvalidArgumentException;
+
+/**
+ * MailSubjectContains
+ *
+ * @internal
+ */
+class MailSubjectContains extends MailConstraintBase
+{
+    /**
+     * Checks constraint
+     *
+     * @param mixed $other Constraint check
+     * @return bool
+     */
+    public function matches($other): bool
+    {
+        if (!is_string($other)) {
+            throw new InvalidArgumentException(
+                'Invalid data type, must be a string.'
+            );
+        }
+        $messages = $this->getMessages();
+        foreach ($messages as $message) {
+            $subject = $message->getOriginalSubject();
+
+            $other = preg_quote($other, '/');
```

Comment:
The current assertion method documentation doesn't imply that regexp patterns are supported. Is using regex an intentional feature or should this constraint be using strpos() instead?

Reply:
´assertMailSubjectContains´ would work fine with `strpos()`, I can change that tomorrow if everyone is fine with that.